File: server/models/projectiles/Projectile.js
const Matter = require('matter-js');
const BaseObject = require('../BaseObject');


class Projectile extends BaseObject {
    constructor(x = 0, y = 0, xSpeed = 0, ySpeed = 0, radius, type, speed, damage, id, agentId) {
        super(id);
        this.bounds = Matter.Bodies.circle(x, y, radius);
        this.velocity = {};
        this.velocity.x = xSpeed;
        this.velocity.y = ySpeed;
        this.speed = speed;
        this.type = type;
        this.agentId = agentId;
        this.damage = damage;
    }
}

module.exports = Projectile;

File: core/src/com/mygdx/game/model/projectile/Projectile.kt
package com.mygdx.game.model.projectile

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Sprite
import com.badlogic.gdx.math.Circle
import com.badlogic.gdx.math.Vector2

abstract class Projectile(
        x: Float = 0f,
        y: Float = 0f,
        xSpeed: Float = 0f,
        ySpeed: Float = 0f,
        texture: Texture,
        radius: Float,
        val speed: Float) {

    val bounds: Circle = Circle()
    val sprite: Sprite
    val velocity: Vector2
    var agentId = ""

    var justFired = false

    init {
        bounds.radius = radius
        bounds.x = x
        bounds.y = y
        sprite = Sprite(texture)
        sprite.setSize(radius * 2, radius * 2)
        sprite.setPosition(x - bounds.radius, bounds.radius - 4f)
        velocity = Vector2(xSpeed, ySpeed)
    }

    fun setPosition(x: Float, y: Float) {
        bounds.x = x
        bounds.y = y
        sprite.setPosition(x - bounds.radius, y - bounds.radius)
    }
}